groups/reconcile.go

```diff
@@ -202,30 +202,66 @@ func readConfig(configFilePath string, confirmChanges bool) error {
 	return err
 }
 
-func readGroupsConfig(configFilePath string, groupsConfigFilePath string, config *GroupsConfig) error {
+func readGroupsConfig(configFilePath string, groupsFileName string, config *GroupsConfig) error {
 	if *verbose {
-		log.Printf("reading group file %s", groupsConfigFilePath)
+		log.Printf("reading group file %s", groupsFileName)
 	}
-	var content []byte
-	var err error
-	groupsUrl, err := url.ParseRequestURI(groupsConfigFilePath)
+
+	groupsURL, err := url.ParseRequestURI(groupsFileName)
 	if err == nil {
-		// We have a URL, so try reading from it
-		if len(groupsUrl.Host) > 0 {
-			if content, err = readFromUrl(groupsUrl); err != nil {
-				return fmt.Errorf("error reading groups config URL %s: %v", groupsUrl, err)
-			}
+		if len(groupsURL.Host) > 0 {
+			// We have a URL, so try reading from it
+			return readGroupsConfigFromURL(groupsURL, config)
 		}
 	} else {
-		// We don't have a URL, we have a file path, so try reading from the file
-		path := groupsConfigFilePath
-		// Interpret relative path as relative to config file
-		if !filepath.IsAbs(path) {
-			path = filepath.Join(filepath.Dir(configFilePath), path)
-		}
-		if content, err = ioutil.ReadFile(path); err != nil {
-			return fmt.Errorf("error reading groups config file %s: %v", groupsConfigFilePath, err)
+		// We don't have a URL, we have a file name
+		// so recursively read all files that have
+		// the filename as groupsFileName
+		return readMultipleGroupsConfig(configFilePath, groupsFileName, config)
+	}
+	return nil
+}
+
+// readMultipleGroupsConfig uses configFilePath's dir as the root dir and
+// recursively walks through all dirs and files. It reads the GroupsConfig
+// from all files that have the name groupsFileName and adds the groups in
+// each GroupsConfig to config.Groups.
+func readMultipleGroupsConfig(configFilePath string, groupsFileName string, config *GroupsConfig) error {
+	rootDir := filepath.Dir(configFilePath)
```

Comment:
nit: I think it would be cleaner if this function took `rootDir` directly, and the work of deriving it was moved elsewhere